src/refactor_nrepl/ns/ns_parser.clj

```diff
@@ -55,19 +54,19 @@
   `(let [~'libspecs (rest (core/get-ns-component ~ns-form ~key))]
      ~@body))
 
-(defn- extract-required-macros [libspec-vector]
+(defn extract-required-macros [libspec-vector]
   {:ns (first libspec-vector)
    :require-macros (drop-while #(not= :refer %) libspec-vector)})
 
 (defn- extract-libspecs [ns-form]
   (mapcat identity
           [(with-libspecs-from ns-form :require
-             (->> libspecs
+             (->> ns-form
```

Comment:
I assume that `with-libspecs-from` is establishing this binding you removed. Not sure if you can teach `clj-kondo` about this. // @borkdude

Reply:
You need to add a .clj-kondo/config.edn with:

```
{:linters {:unresolved-symbol {:exclude [(refactor-nrepl.ns.ns-parser/with-libspecs-from [libspecs])]}}}
```